1-org/modules/centralized-logging/main.tf

```diff
@@ -118,52 +117,115 @@ resource "time_sleep" "wait_sa_iam_membership" {
+}
+
+#----------------------------------------------#
+# Send logs to Log project - Internal Log sink #
+#----------------------------------------------#
+
+module "internal_project_log_export" {
+  source  = "terraform-google-modules/log-export/google"
+  version = "~> 7.8"
+  count   = var.project_options != null ? 1 : 0
+
+  destination_uri        = "logging.googleapis.com/projects/${var.logging_destination_project_id}/locations/${var.project_options.location}/buckets/${coalesce(var.project_options.log_bucket_id, "AggregatedLogs")}"
+  filter                 = var.project_options.logging_sink_filter
+  log_sink_name          = "${coalesce(var.project_options.logging_sink_name, local.logging_sink_name_map["prj"])}-la"
+  parent_resource_id     = var.logging_destination_project_id
+  parent_resource_type   = "project"
+  unique_writer_identity = true
+}
+
+module "destination_aggregated_logs" {
+  source  = "terraform-google-modules/log-export/google//modules/logbucket"
+  version = "~> 7.8"
+  count   = var.project_options != null ? 1 : 0
+
+  project_id                    = var.logging_destination_project_id
+  name                          = coalesce(var.project_options.log_bucket_id, "AggregatedLogs")
+  log_sink_writer_identity      = module.internal_project_log_export[0].writer_identity
+  location                      = var.project_options.location
+  enable_analytics              = var.project_options.enable_analytics
+  linked_dataset_id             = var.project_options.linked_dataset_id
+  linked_dataset_description    = var.project_options.linked_dataset_description
+  retention_days                = var.project_options.retention_days
+  grant_write_permission_on_bkt = false
+}
+
+#-------------------------------------------------#
+# Send logs to Log project - update _Default sink #
+#-------------------------------------------------#
+
+data "google_client_config" "default" {
 }
 
-#------------------------------------------------------------------#
-# Log Bucket Service account IAM membership for log_export_billing #
-#------------------------------------------------------------------#
-resource "google_project_iam_member" "logbucket_sink_member_billing" {
-  count = var.enable_billing_account_sink == true && var.logbucket_options != null ? 1 : 0
+resource "terracurl_request" "exclude_external_logs" {
+  count = var.project_options != null ? 1 : 0
+
+  name           = "exclude_external_logs"
+  url            = "https://logging.googleapis.com/v2/projects/${var.logging_destination_project_id}/sinks/_Default?updateMask=exclusions"
+  method         = "PUT"
+  response_codes = [200]
+  headers = {
+    Authorization = "Bearer ${data.google_client_config.default.access_token}"
+    Content-Type  = "application/json",
+  }
+  request_body = <<EOF
+{
+  "exclusions": [
+    {
+      "name": "exclude_external_logs",
+      "filter": "-logName : \"/${var.logging_destination_project_id}/\""
```

Comment:
I'm not certain if I understand the outcome of this code, so allow me to double-check...

1. There is an aggregated log sink to send audit logs from all projects to `logging_destination_project_id`.
2. There is an additional log sink to send all logs from that project sink to the `aggregatedLogs` bucket sink
3. The exclusion removes logs that originated in `logging_destination_project_id` from #2. This prevents double storing logs from the log project in the `aggregatedLogs` bucket.

Reply:
> 1. There is an aggregated log sink to send audit logs from all projects to logging_destination_project_id.

the aggregate sink send logs from the root of the deploy and all its children.
if the foundation is deployed at org level the sink will route:
- the logs from the organization
- the logs from all folders
- the log form all projects

> 2. There is an additional log sink to send all logs from that project sink to the aggregatedLogs bucket sink

Correct

> 3. The exclusion removes logs that originated in logging_destination_project_id from feat: Adding org module to example #2. This prevents double storing logs from the log project in the aggregatedLogs bucket.

the exclusion is removing logs that does not contain the logging project ID in the logName from been routed to the `_Default` log bucket so that the _Default log bucket only contains logs generate in its own project(the loggong project)
the _Required log bucket already excludes log from outside of the project

Reply:
Got it, thanks for confirming.